lib/api/apiUtils/object/BackendInfo.js

```diff
@@ -50,6 +52,14 @@ export class BackendInfo {
         if (Object.keys(config.restEndpoints).indexOf(requestEndpoint) < 0) {
             log.trace('requestEndpoint does not match config restEndpoints',
                 { requestEndpoint });
+            if (config.backends.data === 'mem' ||
```

Comment:
as discussed, please separate logic so we have (1) a function checking whether the requestEndpoint is in the restEndpoints, (2) a function checking whether config.backends.data is mem or file and (3) a function checking whether the value of the requestEndpoint in the restEndpoints is valid.